File: music-commons-api/src/main/java/net/transgressoft/commons/music/playlist/UnknownAttributeException.java
package net.transgressoft.commons.music.playlist;

import net.transgressoft.commons.query.EntityAttribute;
import net.transgressoft.commons.query.QueryEntity;

public class UnknownAttributeException extends RuntimeException {

    public UnknownAttributeException(EntityAttribute<?> entityAttribute, Class<? extends QueryEntity> queryEntityClass) {
        super(String.format("Unknown attribute %s provided for %s", entityAttribute, queryEntityClass.getName()));
    }
}

File: music-commons-api/src/main/kotlin/net/transgressoft/commons/music/playlist/UnknownAttributeException.kt
package net.transgressoft.commons.music.playlist

import net.transgressoft.commons.query.EntityAttribute
import net.transgressoft.commons.query.QueryEntity

class UnknownAttributeException(entityAttribute: EntityAttribute<*>, queryEntityClass: Class<out QueryEntity>) : RuntimeException(
    String.format("Unknown attribute %s provided for %s", entityAttribute, queryEntityClass.name)
)